torch/_subclasses/fake_tensor.py

```diff
@@ -1937,7 +1937,9 @@ def go(t: object, real_t: Tensor) -> None:
                 # If a data-dependent op is used in a decomposition, we
                 # may need to get the unbacked settings "early"
                 # TODO: Is this really needed?
-                compute_unbacked_bindings(self.shape_env, fake_out, peek=True)
+                compute_unbacked_bindings(
+                    self.shape_env, fake_out, peek=False, real_value=real_out
+                )
```

Comment:
The `peek=False` I'm pretty sure is wrong, but I don't know any other way of clearing fresh symbols?

This is dealing with the case where the created symbool `Eq(u1, 1)` specializes to False, and we can't find an unbacked binding for u1 on the specialized value.